prov/net/src/xnet_progress.c

```diff
@@ -778,8 +778,14 @@ void xnet_progress_rx(struct xnet_ep *ep)
 			ret = ep->cur_rx.handler(ep);
 		}
 
-		if (OFI_SOCK_TRY_SND_RCV_AGAIN(-ret))
+		if (OFI_SOCK_TRY_SND_RCV_AGAIN(-ret)) {
+			xnet_update_pollflag(ep, POLLIN, true);
+			return;
+		} else if (ret == -OFI_EINPROGRESS_URING) {
+			assert(xnet_io_uring);
+			xnet_update_pollflag(ep, POLLIN, false);
```

Comment:
Where did the xnet_update_pollflag() calls above come from?  Or why are these needed here now and not before?